tests/test_runs/test_run.py

```diff
@@ -66,3 +70,55 @@ def test_tagging(self):
         run.remove_tag(tag)
         run_list = openml.runs.list_runs(tag=tag)
         self.assertEqual(len(run_list), 0)
+
+    def _test_run_obj_equals(self, run, run_prime):
+        for dictionary in ['evaluations', 'fold_evaluations', 'sample_evaluations']:
+            if getattr(run, dictionary) is not None:
+                self.assertDictEqual(getattr(run, dictionary), getattr(run_prime, dictionary))
+            else:
+                # should be none or empty
+                other = getattr(run_prime, dictionary)
+                if other is not None:
+                    self.assertDictEqual(other, dict())
+
+        numeric_part = np.array(np.array(run.data_content)[:, 0:-2], dtype=float)
+        numeric_part_prime = np.array(np.array(run_prime.data_content)[:, 0:-2], dtype=float)
+        string_part = np.array(run.data_content)[:, -2:]
+        string_part_prime = np.array(run_prime.data_content)[:, -2:]
+        # JvR: Python 2.7 requires an almost equal check, rather than an equals check
+        np.testing.assert_array_almost_equal(numeric_part, numeric_part_prime)
+        np.testing.assert_array_equal(string_part, string_part_prime)
+
+        if run.trace_content is not None:
+            numeric_part = np.array(np.array(run.trace_content)[:, 0:-2], dtype=float)
+            numeric_part_prime = np.array(np.array(run_prime.trace_content)[:, 0:-2], dtype=float)
+            string_part = np.array(run.trace_content)[:, -2:]
+            string_part_prime = np.array(run_prime.trace_content)[:, -2:]
+            # JvR: Python 2.7 requires an almost equal check, rather than an equals check
+            np.testing.assert_array_almost_equal(numeric_part,numeric_part_prime)
+            np.testing.assert_array_equal(string_part, string_part_prime)
+
+    def test_to_from_filesystem_vanilla(self):
+        model = DecisionTreeClassifier(max_depth=1)
+        task = openml.tasks.get_task(119)
+        run = openml.runs.run_model_on_task(task, model)
+
+        cache_path = os.path.join(self.workdir, 'runs', str(random.getrandbits(128)))
+        os.makedirs(cache_path)
+        run.to_filesystem(cache_path)
+
+        run_prime = openml.runs.OpenMLRun.from_filesystem(cache_path)
+        self._test_run_obj_equals(run, run_prime)
+
+    def test_to_from_filesystem_search(self):
+        model = GridSearchCV(estimator=DecisionTreeClassifier(), param_grid={"max_depth": [1, 2, 3, 4, 5]})
+
+        task = openml.tasks.get_task(119)
+        run = openml.runs.run_model_on_task(task, model)
+
+        cache_path = os.path.join(self.workdir, 'runs', str(random.getrandbits(128)))
+        os.makedirs(cache_path)
+        run.to_filesystem(cache_path)
+
+        run_prime = openml.runs.OpenMLRun.from_filesystem(cache_path)
+        self._test_run_obj_equals(run, run_prime)
```

Comment:
You should add a check here that the trace is available. The function `_test_run_obj_equals` does not guarantee this.